org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/completion/LSCompletionProposal.java

```diff
@@ -306,10 +323,19 @@ public String getDisplayString() {
 
 	@Override
 	public StyledString getStyledDisplayString() {
-		if (Boolean.TRUE.equals(item.getDeprecated())) {
-			return new StyledString(getDisplayString(), StyleUtil.DEPRECATE);
+		StyledString ss = Boolean.TRUE.equals(item.getDeprecated())
```

Comment:
would you mind doing this part of the PR (which I think adds support for completion label details) in a separate PR? That part should be very easy to review then.

Reply:
Agreed.